### ISSUE TITLE
otlp endpoint: No metric translation by default

### ISSUE THREAD

**Issue by ArthurSens** [prometheus/prometheus#14990]
### Proposal

The `/otlp` endpoint is used to ingest metrics in the OpenTelemetry Line Protocol (OTLP) format. Historically, Prometheus couldn't ingest metrics with UTF-8 characters. Functionality that was added in the most recent versions of Prometheus:
* 2.55.0 accepts UTF-8 but requires feature-flag
* 3.0.0 accepts UTF-8 by default

A few months ago, David Ashpole run a [survey](https://opentelemetry.io/blog/2024/prometheus-compatibility-survey/) that shows clear distinction about preferences on how metrics should look like in Prometheus. On one side, Prometheus veterans like the usual Prometheus naming conventions with underscores and unit suffixes. On the other, a new user base is emerging that comes from the OTel world, and those prefer metric name with dots and without unit suffixes.

In Prometheus, there is also a clear distinction:
* Prometheus veterans will use the traditional ways of ingesting data: Scraping and Prometheus Remote Write
* OTel users will use the OTLP endpoint.

Since Prometheus veterans are expected to use the usual prometheus protocols and not the `/otlp` endpoint, we can make OTel  users happier by not translating anything when data is ingested by the mentioned enpoint.

**Comment by ArthurSens** [comment]
Related PR: https://github.com/prometheus/prometheus/pull/14974

**Comment by juliusv** [comment]
> we can make OTel users happier by not translating anything

As said elsewhere, I predict that their happiness will be temporary until they learn about the concept of a query language and having to work with the metric names in there (annoying escaping + not knowing what unit a metric is in) 😅

**Comment by ArthurSens** [comment]
> As said elsewhere, I predict that their happiness will be temporary until they learn about the concept of a query language and having to work with the metric names in there (annoying escaping + not knowing what unit a metric is in) 😅

That's a pretty valid concern! It's just not the feedback we've been receiving 😬 

I wonder if we can find a sweet middle spot where users can have their metrics unchanged and not struggle with what you just mentioned. More mature toolchain like language server and VS code extensions could maybe help here?

**Comment by juliusv** [comment]
> That's a pretty valid concern! It's just not the feedback we've been receiving 😬

Yep, of course the question is how much of the full metrics usage the average OTel (vs. Prometheus) user would consider.

> I wonder if we can find a sweet middle spot where users can have their metrics unchanged and not struggle with what you just mentioned. More mature toolchain like language server and VS code extensions could maybe help here?

This would be ideal, but I don't see it happening any time soon in a way that'll be easily and ubiquitously accessible to the average person staring at a PromQL expression in a YAML file in whatever editing (or just GitHub) environment they are currently using to read code. And of course it would be even if PromQL was fully unit-aware, so you never have to think about units at all, but I expect that even less soon :)

It seems people at the last DevSummit have decided to go ahead anyway with less (or no) conversions for OTel metrics by default, so I guess that's what will happen for now. I just want to be on the record once again to say that I think people may regret this once they consider more fully how metrics are used in practice.

**Comment by krajorama** [comment]
Hello from the bug scrub!

Work is ongoing in this area.
